# Chapter 1

## 오류의 두 가지 종류
1. 컴파일 오류

    컴파일 단계에서 오류 발생(코드를 잘못 씀)

2. 런타임 오류

    실행 단계에서 오류 발생(DB에 데이터가 없음, 0으로 나눔 등)

## Print
```python
print("Hello python", 123, end="!!!!!")
# 매개변수에 *values: object 이므로 튜플을 받는다.
# 그러므로 print는 여러개의 argument를 튜플의 형태로 받는다.
# end = "\n" 였는데 \n 대신에 !!!!!로 바꿨기 때문에 개행 하지 않는다

print("진짜로 개행 안 하는지 확인") # ㅇㅇ 개행 안 했네

# print(end="!!!", values=("Hello python", "test")) 이런 개념도 있기는 하다
```

기본적으로 여러개를 받을 수 있다

## Type

기본 타입 중 string 타입에는 여러 함수를 적용할 수 있다
```python
number = 10
name = "강수림수림" # 중요 : 여러 함수  다른 건 그냥 그런데 얘가 좀사용 가능
isOpen = True

print(number)
# print(name_) # 오류 앞 라인까지는 실행 된다
print(name)
print(isOpen)

# print(name.index("숭"))  # exception
# print(name.find("숭"))   # return -1

print(name.index("수",2))
print(name.find("수"))

print("이름은: {name}, 숫자: {number}, 열림: {isOpen}".format(name=name, number=number, isOpen=isOpen))
print(f"이름은: {name}, 숫자: {number}, 열림: {isOpen}") # f 포매팅 : 정처기에 나온 적 있음, 여기서는 변수를 넣었지만 함수 등도 사용 가능
address = """부
산
동
래
구
중앙대로""" # 여러 줄로 나누어 쓸 수 있음
print(address)
```

## Collection

### 컬렉션의 종류

1. set
2. list
3. dict
4. tuple

### 실행 예
```python
set1 = set()
list1 = list()
dict1 = dict()
tuple1 = tuple()

print(type(set1)) # 자료형을 확인하는 type 함수

list2 = []
dict2 = {}
dict3 = {
    "n":0
}
tuple2 = ()
tuple3 = (10) # 10 <- 값이 하나라서 ()를 연산 기호로 본다
tuple4 = (10, ) # ,가 들어가면 그 때 부터 튜플로 본다
set2 = {0}


print(list2) # []
print(dict2) # {}
print(tuple2) # ()
print(set2) # {0}
print(dict3) # {'n': 0}

print(type(()))
```